lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,76 @@
+
+### Projects
+
+Network resources will be created within a ``control-plane`` project.  We will be utilizing shared networks, which must be created by a user with the ``cloud_admin`` role on that project. Compute resources for control plane services will be created within the ``control-plane`` project.  Nodepool itself will be configured to use a specific ``nodepool`` project for creating its instances with interfaces attached to a shared network that was created on the ``control-plane`` project.
+
+### Networks
+
+We will create two separate internal networks and hosts will be given interfaces on them depending on the role of said host.
+
+ * ``control-plane 192.168.10.0/24`` - This network will be used for communication between all of our production services, as well as automation between the bastion host and our producion servers.  As per our current infrastructure, the following static hosts will have interfaces on this network and internal DNS updated accordingly:
+
+ - bastion.bonncyi-internal.portbleu.com
+ - logs.bonnyci-internal.portbleu.com
+ - merger01.bonnyci-internal.portbleu.com
+ - nodepool.bonncyi-internal.portbleu.com
+ - zuul.bonnyci-internal.portbleu.com
+
+ * ``nodepool 10.0.0.0/8`` - This is an shared network that will be used by nodepool slaves. None of the static service hosts within the control plane will receive an interface on this network.
+
+Our cloud is also configured with an external network named ``external`` that provides us currently with 30 floating IPs.  These will be assigned to hosts depending on their need, see [Floating IPs](#floating-ips).
+
+### Routers
+
+To allow connectivity between the two isolated networks, a single L3 router will be created.  The router will have a router interface attached for both the ``control-plane`` and ``nodepool`` networks.  Additionally, it will have its router gateway set to provide outbound connectivity through the external network and allow for floating IP association.
+
+### Floating IPs
+
+Some of the hosts running on the control plane will need to be accessed from the outside world for various reasons.  Public addresses for floating IPs will be allocated from the ``external`` network that is available to our cloud, currently providing addresses from the 169.44.171.64/27 range, though our quota currently allows for 50. Each of the following hosts will also be assigned a floating IP from the external network that will be publically accessible on specific ports (see [Security Groups](#security-groups)). Public DNS hostnames will be updated accordingly:
```

Comment:
Right, our quota gives us access to more IPs than we actually have. I think we should be OK with what we have currently. We should only really need the 3 for the services mentioned here for now, plus whatever we allow developers to use.  I seem to remember hearing something about VPN access on the BB onboarding meeting so its possible we may not even need floaters for dev access